fix subscription list tint

EricBAndrews opened the pull request:
<!--
Thank you for opening a pull request!

We assume you have read CONTRIBUTING.md. If you have not, do not be surprised if your PR is rejected for reasons listed therein.

Please note that if your PR does not address an issue that was assigned to you, you are still welcome to open it; however, it will not receive merge priority and may be rejected for scope reasons.
-->

## Issues

- closes #2216 

## Description

- Removes the default tint from our custom `Form`--as far as I can tell it didn't actually do anything other than prevent the form tint from being locally overridden.
- Tint the subscription list `.themedPrimary`
- Force a `.themedSecondary` tint on the `MediaView` spinner so it always renders as gray rather than picking up parent tint (this is consistent with other spinners in the app)

EricBAndrews committed:
fix subscription list tint

Sjmarf commented:
The icons for destructible actions are no longer red with this fix. Would supporting that be possible?

Sjmarf commented:
Also, the toggles in Settings are now green - do we want that?

EricBAndrews committed:
fix tint

EricBAndrews committed:
remove extra subscriptionlist tint

EricBAndrews committed:
conditional tint

EricBAndrews commented:
Nice catch. I've modified `Form` to take `tint` as an optional parameter.

Re: destructive actions, it looks like tinting a Form really aggressively applies that tint to everything inside it--we didn't encounter this before because we pretty much only use form in settings. I've updated `MenuButton` to force a `.themedNegative` tint, which gets the buttons correctly colored. I don't think this has any side effects anywhere else, since we were already coloring them that way.

Sjmarf reviewed (approved):
Nice 👍

EricBAndrews committed:
Merge branch 'dev' into eric/subscription-list-tint
